includes/class.llms.template.loader.php

```diff
@@ -332,50 +332,79 @@ public function restricted_by_lesson_prerequisite( $info ) {
 
-		$membership_id = $info['restriction_id'];
+		$membership_ids = $info['restriction_id'];
 
 		// Do nothing if we don't have a membership id.
-		if ( ! empty( $membership_id ) && is_numeric( $membership_id ) ) {
-
-			// Instantiate the membership.
-			$membership = new LLMS_Membership( $membership_id );
+		if ( ! empty( $membership_ids ) && ( is_array( $membership_ids ) || is_numeric( $membership_ids ) ) ) {
 
 			$msg      = '';
 			$redirect = '';
 
-			if ( 'yes' === $membership->get( 'restriction_add_notice' ) ) {
+			// Check if we're dealing with an array or a numeric for single membership restriction.
+			if ( ( is_array( $membership_ids ) && 1 === count( $membership_ids ) ) || is_numeric( $membership_ids ) ) {
 
-				$msg = $membership->get( 'restriction_notice' );
+				// Get the membership.
+				$membership = new LLMS_Membership( is_array( $membership_ids ) ? $membership_ids[0] : $membership_ids );
 
-			}
+				if ( 'yes' === $membership->get( 'restriction_add_notice' ) ) {
+
+					$msg = $membership->get( 'restriction_notice' );
 
-			// Get the redirect based on the redirect type (if set).
-			switch ( $membership->get( 'restriction_redirect_type' ) ) {
+				}
+
+				// Get the redirect based on the redirect type (if set).
+				switch ( $membership->get( 'restriction_redirect_type' ) ) {
+
+					case 'custom':
+						$redirect = $membership->get( 'redirect_custom_url' );
+						break;
+
+					case 'membership':
+						$redirect = get_permalink( $membership->get( 'id' ) );
+						break;
+
+					case 'page':
+						$redirect = get_permalink( $membership->get( 'redirect_page_id' ) );
+						// Make sure to print notices in wp pages.
+						$redirect = empty( $msg ) ? $redirect : add_query_arg(
+							array(
+								'llms_print_notices' => 1,
+							),
+							$redirect
+						);
+						break;
+				}
+			} else {
 
-				case 'custom':
-					$redirect = $membership->get( 'redirect_custom_url' );
-					break;
+				$restricted_memberships = '';
+				$count                  = 0;
+				$length                 = count( $membership_ids );
 
-				case 'membership':
-					$redirect = get_permalink( $membership->get( 'id' ) );
-					break;
+				foreach ( $membership_ids as $membership_id ) {
 
-				case 'page':
-					$redirect = get_permalink( $membership->get( 'redirect_page_id' ) );
-					// Make sure to print notices in wp pages.
-					$redirect = empty( $msg ) ? $redirect : add_query_arg(
-						array(
-							'llms_print_notices' => 1,
-						),
-						$redirect
-					);
-					break;
+					$restricted_memberships .= do_shortcode( '[lifterlms_membership_link id="' . $membership_id . '"]' );
+					$count++;
+
+					// Adding `, ` or ` or ` depending on the number of memberships.
+					if ( $count < $length - 1 ) {
+						$restricted_memberships .= ', ';
+					} elseif ( $count === $length - 1 ) {
+						$restricted_memberships .= __( ' or ', 'lifterlms' );
+					}
+				}
+
+				// Translators: %s = Membership links.
+				$msg = sprintf(
```

Comment:
We want to keep the messaging as-is if there is only one membership they need to be a member of.